src/app/pages/student/student.component.ts

```diff
@@ -23,9 +24,11 @@ export class StudentComponent implements OnInit {
 
   constructor(private studentService: StudentService, private dialog: MatDialog) {}
 
-  async ngOnInit() {
+  ngOnInit() {
     this.dataSource.sort = this.sort;
+
     this.subCollector.add(
```

Comment:
The main fix : prefetch the students list, because the previous behavior (with apollo watchQuery) wasn't notified about the list change, when coming back from "create/update class" routes